### PR TITLE
upgrade to new visual studio 2017 project model

### PR DESCRIPTION
**Pull request by OrbintSoft**
I upgraded csproj to new visual studio 2017 project model
All references now depends only by .nuget, and uses latest .nuget project model
Now should be easy to target to .net core or .net standard

I also had to upgrade framework dependences to .NET 4.5, maybe can be possible to create a multi target with .NET 3.5 support, but it's more easy to deprecate the old framework

### CODE REVIEW

**Comment by danesparza** [comment]
I really appreciate the commit -- I'll try to get this into an accepted state shortly.  Unfortunately AssemblyInfo.cs is no longer present by default (and that's what was used for version patching).  I'll need to research how to sync version information with the new project format.  😞

**Comment by OrbintSoft** [comment]
AssemblyInfo still exist you can find in /obj folder, but some properties now are stored in .csproj and auto generated at build time.

I removed them because if duplicated, the build generates an error.

You can disable auto generated proprieties adding these to .csproj: 

<GenerateAssemblyInfo>false</GenerateAssemblyInfo>


I was thinking if keep both .csproj for backward compatibility, one that targets .net 3.5 and the other .net standard 2.0 because you can create a multi target .nuget package, but I think that makes more complicated to maintain the library, so I chosen to deprecate NET .3.5.


I upgraded to .NET 4.5 because I was unable to find equivalent libraries on .nuget for 3.5.


I would like to tests the new .nuget package before add .net standard support.

Thank you.

**Comment by OrbintSoft** [comment]
Any news? let me know if you need help, I can restore generation AssemblyInfo.cs if that is the only problem.
Sorry to bother you, but I need this library and I am planning to do many contributions, If you have no time to maintain, I'll fork it and release a new .nuget.
Thank you.

File: DomainParser.TestConsole/Program.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@ namespace DomainParser.TestConsole {
     class Program {
         static void Main(string[] args) {
 
+            //What the hell is d: ????  :) :D (*_*)
             TLDRulesCache.RulesFileLocation = @"d:\" + Path.GetFileName(TLDRulesCache.RulesFileLocation);
```
Comment (danesparza) [inline DomainParser.TestConsole/Program.cs:12]: Great catch.  Looks like a contributor accidentally hardcoded the root path for a test.... whoopsie. 

⭐️

**Pull request merged**